Source/UpgradeScripts/v2_10_1RemovePayload.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using SaveUpgradePipeline;
+using UnityEngine;
+
+namespace RP0.UpgradeScripts
+{
+
+    [UpgradeModule(LoadContext.Craft, craftNodeUrl = "PART")]
+    public class v2_10_1RemovePayload : UpgradeScript
+    {
+        public override string Name { get => "RP-1 ComSatPayload and NavSatPayload Upgrader"; }
+        public override string Description { get => "Removes ComSatPayload and NavSatPayload from older Tanks"; }
+        public override Version EarliestCompatibleVersion { get => new Version(2, 0, 0); }
+        protected static Version _targetVersion = new Version(2, 10, 1);
+        public override Version TargetVersion => _targetVersion;
+
+        private TestResult TestRecurse(ConfigNode node)
+        {
+            ConfigNode.Value v = null;
+            if (node.name == "RESOURCE" && (v = node.values.values.Find(s => s.name == "name")) != null)
+            {
+                if (v.value == "ComSatPayload" || v.value == "NavSatPayload")
+                {
+                    return TestResult.Upgradeable;
+                }
+            }
+
+            foreach (ConfigNode n in node.nodes)
+            {
+                var res = TestRecurse(n);
```

Comment:
This won't actually find MFT blocks, because their node name is TANK not RESOURCE, no?

Reply:
(and if that's not your intention here, what _are_ you trying to find by recursing down every node in the PART node?)